iac-extensions/kubernetes/src/test/java/org/sonar/iac/helm/HelmEvaluatorTest.java

```diff
@@ -20,48 +20,88 @@
 package org.sonar.iac.helm;
 
 import com.google.protobuf.InvalidProtocolBufferException;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
 import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 import org.mockito.Mockito;
-import org.sonar.iac.helm.jna.Loader;
-import org.sonar.iac.helm.jna.library.IacHelmLibrary;
+import org.slf4j.event.Level;
+import org.sonar.api.testfixtures.log.LogTesterJUnit5;
+import org.sonar.iac.helm.utils.ExecutableHelper;
 import org.sonarsource.iac.helm.TemplateEvaluationResult;
 
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
 
 class HelmEvaluatorTest {
-  @Test
-  void shouldThrowIfRawEvaluationResultIsEmpty() {
-    // Mock methods that are executed by JNA in runtime
-    var emptyRawResult = Mockito.mock(IacHelmLibrary.EvaluateTemplate_return.ByValue.class);
-    Mockito.when(emptyRawResult.getByteArray()).thenReturn(new byte[0]);
-    var iacHelmLibrary = Mockito.mock(IacHelmLibrary.class);
-    Mockito.when(iacHelmLibrary.evaluateTemplate(any(), any(), any())).thenReturn(emptyRawResult);
-    var helmEvaluator = new HelmEvaluator(iacHelmLibrary);
+  @TempDir
+  File tempDir;
+
+  @RegisterExtension
+  public LogTesterJUnit5 logTester = new LogTesterJUnit5().setLevel(Level.DEBUG);
+  private HelmEvaluator helmEvaluator;
 
+  @BeforeEach
+  void setUp() throws IOException {
+    this.helmEvaluator = new HelmEvaluator(tempDir);
+    this.helmEvaluator.initialize();
+  }
+
+  @Test
+  void shouldThrowIfGoBinaryReturnsNonZero() {
     Assertions.assertThatThrownBy(() -> helmEvaluator.evaluateTemplate("/foo/bar/baz.yaml", "", ""))
       .isInstanceOf(IllegalStateException.class)
-      .hasMessage("Empty evaluation result (serialization failed?)");
+      .hasMessage("sonar-helm-for-iac exited with non-zero exit code: 1");
+
+    Assertions.assertThat(logTester.logs(Level.DEBUG))
+      // TODO: better handling of empty input
```

Comment:
What about this TODO?

Reply:
This was about the line `Expected 2 files, received 3`, when the input is in fact completely empty. But for now I don't want to overcomplicate Go logic to handle such rare cases; I think the messages `Skipping request to read 0 lines` before this line make it a bit more clear